client/components/LearnerViewDirective.js

```diff
@@ -23,12 +23,19 @@ tie.directive('learnerView', [function() {
     template: `
       <div class="tie-exercise-ui-outer">
         <div class="tie-exercise-ui-inner">
-          <div>
-            Stepper:
-            <div ng-repeat="questionId in questionIds track by $index">
-              <span ng-class="{'tie-active-question-index': currentQuestionIndex === $index}">
-                {{$index + 1}}
-              </span>
+          <div class="tie-step-container-outer">
+            <div class="tie-step-container-inner">
+              <div class="tie-step-item" ng-repeat="questionId in questionIds\
+                  track by $index"
+                  ng-click="navigateToQuestion($index)">
+                <div class="tie-step-circle" ng-class="{'tie-step-active':\
+                    currentQuestionIndex === $index, 'tie-step-unlocked':\
+                    unlockedQuestions[$index]}">
+                  <span class="tie-step-text">{{$index + 1}}</span>
+                </div>
+                <div ng-class="{'tie-step-line':\
+                    $index < (questionIds.length - 1)}"></div>
```

Comment:
If you're breaking a tag multi-line, consider moving the closing tag to a separate line. Easier to keep track of indentation that way, since one can eyeball that the `<div>` and `</div>` have the same indentation. Ditto elsewhere.